packages/amplify-appsync-simulator/src/server/operations.ts

```diff
@@ -123,12 +123,9 @@ export class OperationServer {
             variables,
             operationName
           );
-          if (Object.keys(context.appsyncErrors).length) {
-            results.errors = JSON.stringify(context.appsyncErrors);
-          }
-          // Make extensions available at the root level
-          if (results.errors) {
-            results.errors = exposeGraphQLErrors(results.errors);
+          const errors = [...(results.errors || []), ...context.appsyncErrors];
+          if (errors.length) {
```

Comment:
Don't we need a >0 check since the ```errors``` will always be an initialized array and will have a ```length``` property defined.